frontend/test/pytest/test_cuda_integration.py

```diff
@@ -0,0 +1,459 @@
+# Copyright 2022-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""CUDA Integration testing."""
+
+import jax
+import pennylane as qml
+import pytest
+from jax import numpy as jnp
+from numpy.testing import assert_allclose
+
+from catalyst import measure, qjit
+from catalyst.compilation_pipelines import QJIT
+from catalyst.compiler import CompileOptions
+from catalyst.utils.jax_extras import remove_host_context
+
+# This import is here on purpose. We shouldn't ever import CUDA
+# when we are running kokkos. Importing CUDA before running any kokkos
+# kernel polutes the environment and will create a segfault.
+# pylint: disable=import-outside-toplevel
+
```

Comment:
```suggestion
pytest.importorskip("cudaq")
```

Reply:
I can't do this. Even using `pytest.importskip("cudaq")` will import CUDA before running any test. This means that when testing kokkos this file will indeed import cudaq and kokkos tests will segfault. This is why we don't import cudaq here and why we don't import parts of catalyst which import cudaq.